manifests/agent.pp

```diff
@@ -295,9 +297,11 @@
     subscribe => $service_subscribe,
   }
 
-  sensu_agent_entity_validator { $config['name']:
-    ensure    => 'present',
-    namespace => $config['namespace'],
-    provider  => 'sensu_api',
+  if $validate_agent_entity {
```

Comment:
I think you could get rid of this parameter and just use `agent_managed_entity` and if the agent manages the entity, don't use the validator.  This is already how some behavior works like here: https://github.com/sensu/sensu-puppet/blob/5a067b9713434e7e791b6ae887e1580f326efeaa/manifests/agent/annotation.pp#L52